### PR TITLE
Mod in chain generic

### PR DESCRIPTION
**Pull request by fluidvanadium**
This is is legibility PR. Set up some mods that will make moving trait implementations between files easier later. Now the use statements wont get tangled between the trait and the tests.

### CODE REVIEW

File: integration-tests/tests/chain_generic_tests.rs
Comment (Oscar-Pepper) [inline integration-tests/tests/chain_generic_tests.rs:1]: effort has gone into moving this code but not out of the `tests` directory ? this is test framework not a test itself so doesnt belong here

File: darkside-tests/src/chain_generic_tests.rs [hunk 1]
```diff
@@ -42,83 +36,94 @@ proptest! {
 ///   - transparent sends do not work
 ///   - txids are regenerated randomly. zingo can optionally accept_server_txid
 /// these tests cannot portray the full range of network weather.
-impl ConductChain for DarksideEnvironment {
-    async fn setup() -> Self {
-        DarksideEnvironment::new(None).await
-    }
+pub(crate) mod impl_conduct_chain_for_darkside_environment {
```
Comment (Oscar-Pepper) [inline darkside-tests/src/chain_generic_tests.rs:39]: I think symmetry is important, can we name this `darkside_environment` to match `libtonode_environment` which serves the same purpose of including the impl for conduct chain? this mod name seems too long and specific

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]
my comments from previous PRs werent addressed and that code has been moved again but not to the right place

File: darkside-tests/src/chain_generic_tests.rs (hunk 1 above)
Comment (fluidvanadium) [reply darkside-tests/src/chain_generic_tests.rs:39]: well, this mod doesnt include the actual definition of the struct, so it is importantly different from the other

File: integration-tests/tests/chain_generic_tests.rs
Comment (fluidvanadium) [reply integration-tests/tests/chain_generic_tests.rs:1]: i have created a mod. the mod will be easy to move later

**Review by Oscar-Pepper — APPROVED** [review]
i will organise this when we have time, nothing is broken here

**Pull request merged**